mycore-user2/src/main/java/org/mycore/user2/MCRUserServlet.java

```diff
@@ -74,10 +75,96 @@ public class MCRUserServlet extends MCRServlet {
     /** The logger */
     private static final Logger LOGGER = LogManager.getLogger(MCRUserServlet.class);
 
+    private static boolean checkUserIsNotNull(HttpServletResponse res, MCRUser currentUser, String userID) {
+        if (currentUser == null) {
+            String uid = userID == null ? MCRSessionMgr.getCurrentSession().getUserInformation().getUserID() : userID;
+            String msg = MCRTranslation.translate("component.user2.UserServlet.currentUserUnknown", uid);
+            try {
+                res.sendError(HttpServletResponse.SC_FORBIDDEN, msg);
+            } catch (IOException e) {
+                throw new MCRException("Error sending error", e);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean checkUserIsLocked(HttpServletResponse res, MCRUser currentUser) throws IOException {
+        if (currentUser.isLocked()) {
+            String userName = currentUser.getUserID();
+            String msg = MCRTranslation.translate("component.user2.UserServlet.isLocked", userName);
+            res.sendError(HttpServletResponse.SC_FORBIDDEN, msg);
+            return true;
+        }
+        return false;
+    }
+
+    private static boolean checkUserIsDisabled(HttpServletResponse res, MCRUser currentUser) throws IOException {
+        if (currentUser.isDisabled()) {
+            String userName = currentUser.getUserID();
+            String msg = MCRTranslation.translate("component.user2.UserServlet.isDisabled", userName);
+            res.sendError(HttpServletResponse.SC_FORBIDDEN, msg);
+            return true;
+        }
+        return false;
+    }
+
+    private static boolean forbidIfGuest(HttpServletResponse res) throws IOException {
+        if (MCRSessionMgr.getCurrentSession().getUserInformation().getUserID()
+            .equals(MCRSystemUserInformation.getGuestInstance().getUserID())) {
+            String msg = MCRTranslation.translate("component.user2.UserServlet.noGuestAction");
+            res.sendError(HttpServletResponse.SC_FORBIDDEN, msg);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Invoked by editor form user-editor.xed to check for a valid
+     * login user name.
+     */
+    public static boolean checkUserName(String userName) {
+        String realmID = MCRRealmFactory.getLocalRealm().getID();
+
+        // Check for required fields is done in the editor form itself, not here
+        if ((userName == null) || (realmID == null)) {
+            return true;
+        }
+
+        // In all other cases, combination of userName and realm must not exist
+        return !MCRUserManager.exists(userName, realmID);
+    }
+
+    private static void updateUserOwner(String userName, String realmID, MCRUser user) {
```

Comment:
This method name is confusing because it updates the actual user and has nothing todo with the owner.